HW6/app_001.py

```diff
@@ -0,0 +1,210 @@
+from datetime import datetime
+from typing import List
+
+import databases
+import sqlalchemy
+from fastapi import FastAPI
+from pydantic import BaseModel, Field
+from sqlalchemy import create_engine, Boolean
+
+DATABASE_URL = "sqlite:///mydatabase.db"
+
+database = databases.Database(DATABASE_URL)
+
+metadata = sqlalchemy.MetaData()
+
+users = sqlalchemy.Table(
+    "users", metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("firstname", sqlalchemy.String(50)),
+    sqlalchemy.Column("lastname", sqlalchemy.String(50)),
+    sqlalchemy.Column("email", sqlalchemy.String(128)),
+    sqlalchemy.Column("password", sqlalchemy.String(255)),
+)
+
+products = sqlalchemy.Table(
+    "products", metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("name", sqlalchemy.String(50)),
+    sqlalchemy.Column("description", sqlalchemy.String(1000)),
+    sqlalchemy.Column("price", sqlalchemy.Float),
+)
+
+orders = sqlalchemy.Table(
+    "orders", metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("user_id", sqlalchemy.Integer, sqlalchemy.ForeignKey('users.id')),
+    sqlalchemy.Column("product_id", sqlalchemy.Integer, sqlalchemy.ForeignKey('products.id')),
+    sqlalchemy.Column("date", sqlalchemy.DateTime),
+    sqlalchemy.Column("status", sqlalchemy.Boolean),
+)
+
+engine = create_engine(DATABASE_URL, connect_args={"check_same_thread": False})
+
+metadata.create_all(engine)
+
+app = FastAPI()
+
+class UserIn(BaseModel):
+    firstname: str = Field(title="First Name", max_length=50)
+    lastname: str = Field(title="Last Name", max_length=50)
+    email: str = Field(itle="Email", max_length=128)
+    password: str = Field(title="Password", max_length=255)
+
+class User(BaseModel):
+    id: int
+    firstname: str = Field(title="First Name", max_length=50)
+    lastname: str = Field(title="Last Name", max_length=50)
+    email: str = Field(itle="Email", max_length=128)
+    # password: str = Field(title="Password", max_length=255)
+
+class ProductIn(BaseModel):
+    name: str = Field(title="Name", max_length=50)
+    description: str = Field(default=None, title="Description", max_length=1000)
+    price: float = Field(title="Price", gt=0, le=1000000)
+
+class Product(BaseModel):
+    id: int
+    name: str = Field(title="Name", max_length=50)
+    description: str = Field(default=None, title="Description", max_length=1000)
+    price: float = Field(title="Price", gt=0, le=1000000)
```

Comment:
Модели почти идентичны, значит можно одну наследовать от другой
```suggestion
class Product(ProductIn):
    id: int
```